Description: Adjust prey-config.py according to the Debian customization.
 Application modules are installed in /usr/lib/prey/.
 Configuration is stored in /etc/prey/config.
 Disable guest account creation for security reasons.
 Don't touch crontab files, the only one installed is managed by cron
 daemon and it is located in /etc/cron.d.
Author: Alessio Treglia <alessio@debian.org>
---
 platform/linux/prey-config.glade |    9 +++++--
 platform/linux/prey-config.py    |   49 +++------------------------------------
 2 files changed, 11 insertions(+), 47 deletions(-)

--- prey.orig/platform/linux/prey-config.py
+++ prey/platform/linux/prey-config.py
@@ -41,11 +41,10 @@ _ = gettext.gettext
 # vars and such
 ################################################
 
-PREY_PATH = '/usr/share/prey'
-CONFIG_FILE = PREY_PATH + '/config'
+PREY_PATH = '/usr/lib/prey'
+CONFIG_FILE = '/etc/prey/config'
 CONTROL_PANEL_URL = 'http://control.preyproject.com'
 CONTROL_PANEL_URL_SSL = 'https://control.preyproject.com'
-GUEST_ACCOUNT_NAME = 'guest_account'
 VERSION = os.popen("cat " + PREY_PATH + "/version 2> /dev/null").read().strip().replace('version=', '').replace("'",'')
 
 PAGES = ['report_options', 'control_panel_options', 'new_user', 'existing_user', 'existing_device', 'standalone_options']
@@ -247,7 +246,6 @@ class PreyConfigurator(object):
 				button_apply.grab_default()
 
 	def ensure_visible(self,widget,event): #ensure the widget focused is visible in the scroll window
-		self.get('delay').set_name('delay')
 		self.get('extended_headers').set_name('extended_headers')
 		widget_name = widget.get_name()
 		scrollwindow = self.get('main_settings_scrollwindow')
@@ -263,10 +261,7 @@ class PreyConfigurator(object):
 
 		# for now we know there are only two possible hidden widgets so we scroll all the way up or all the way down
 		# if we add options to this page we will have to scroll differently
-		if widget_name == 'delay':
-			#scroll to top
-			port_vadjust.set_value(0)
-		elif widget_name == 'extended_headers':
+		if widget_name == 'extended_headers':
 			#scroll to bottom
 			port_vadjust.set_value(internal_height - port_height)
 
@@ -293,7 +288,7 @@ class PreyConfigurator(object):
 			return True
 
 	def is_config_writable(self):
-		command = 'if [ ! -w "'+PREY_PATH+'/config" ]; then echo 1; fi'
+		command = 'if [ ! -w "%s" ]; then echo 1; fi' % CONFIG_FILE
 		no_access = os.popen(command).read().strip()
 		if no_access == '1':
 			self.show_alert(_("Unauthorized"), _("You don't have access to manage Prey's configuration. Sorry."), True)
@@ -306,12 +301,8 @@ class PreyConfigurator(object):
 
 	def get_current_settings(self):
 
-		self.current_delay = os.popen("crontab -l | grep prey | cut -c 3-4").read()
-		if not self.current_delay: self.current_delay = 20
-
 		self.current_auto_connect = self.get_setting('auto_connect')
 		self.current_extended_headers = self.get_setting('extended_headers')
-		self.current_guest_account = self.guest_account_exists()
 
 		self.current_lang = self.get_setting('lang')
 		self.current_check_url = self.get_setting('check_url')
@@ -324,29 +315,7 @@ class PreyConfigurator(object):
 		self.current_smtp_server = self.get_setting('smtp_server')
 		self.current_smtp_username = self.get_setting('smtp_username')
 
-	def guest_account_exists(self):
-		result = os.popen('id ' + GUEST_ACCOUNT_NAME + ' 2> /dev/null').read()
-		if result.find("uid"):
-			return False
-		else:
-			return True
-
-	def toggle_guest_account(self, enabled):
-		if enabled:
-			# create user and leave password blank
-			os.system("useradd -m " + GUEST_ACCOUNT_NAME + "; passwd -d " + GUEST_ACCOUNT_NAME)
-			# Authorize login with no passwords in gdm
-			os.system("sed -i 's/PasswordRequired=false/#PasswordRequired=false/' /etc/gdm/gdm.conf")
-			# Authorize login with no passwords in pam
-			os.system("sed -i 's/nullok_secure/nullok/' /etc/pam.d/common-auth")
-		else:
-			os.system("userdel -r " + GUEST_ACCOUNT_NAME)
-			os.system("sed -i 's/#PasswordRequired=false/PasswordRequired=false/' /etc/gdm/gdm.conf")
-			os.system("sed -i 's/nullok/nullok_secure/' /etc/pam.d/common-auth")
-
 	def display_real_settings(self):
-		self.get('delay').set_value(int(self.current_delay))
-		self.get('guest_account').set_active(self.current_guest_account)
 
 		if self.current_auto_connect == 'y':
 			self.get('auto_connect').set_active(True)
@@ -399,15 +368,6 @@ class PreyConfigurator(object):
 		self.save('auto_connect', self.checkbox('auto_connect'))
 		self.save('extended_headers', self.checkbox('extended_headers'))
 
-		if((self.checkbox('guest_account') == 'y') != self.current_guest_account):
-			self.toggle_guest_account(self.checkbox('guest_account') == 'y')
-
-		# check and change the crontab interval
-		new_delay = self.get('delay').get_value_as_int()
-		if new_delay != int(self.current_delay):
-			# print 'Updating delay in crontab...'
-			os.system('(crontab -l | grep -v prey; echo "*/'+str(new_delay)+' * * * * /usr/share/prey/prey.sh > /var/log/prey.log") | crontab -')
-
 		if self.check_if_configured == False:
 			self.show_alert(_("All good."), _("Configuration saved. Remember you still need to set up your posting method, otherwise Prey won't work!"))
 		else:
@@ -590,7 +550,6 @@ class PreyConfigurator(object):
 		self.pages = builder.get_object("reporting_mode_tabs")
 		self.root = builder
 
-		self.get('delay').grab_focus()
 		about = self.get('about_prey_config')
 		about.set_version(VERSION)
 		self.display_real_settings()
--- prey.orig/platform/linux/prey-config.glade
+++ prey/platform/linux/prey-config.glade
@@ -93,13 +93,13 @@
                                 <property name="has_focus">True</property>
                                 <property name="is_focus">True</property>
                                 <property name="editable">False</property>
+                                <property name="sensitive">False</property>
                                 <property name="invisible_char">&#x2022;</property>
                                 <property name="caps_lock_warning">False</property>
                                 <property name="adjustment">delay_adjustment</property>
                                 <property name="snap_to_ticks">True</property>
                                 <property name="numeric">True</property>
                                 <property name="wrap">True</property>
-                                <signal name="focus_in_event" handler="ensure_visible"/>
                               </object>
                               <packing>
                                 <property name="x">70</property>
@@ -111,6 +111,7 @@
                                 <property name="width_request">250</property>
                                 <property name="height_request">30</property>
                                 <property name="visible">True</property>
+                                <property name="sensitive">False</property>
                                 <property name="xalign">0</property>
                                 <property name="label" translatable="yes" comments="Mnemonic: Yes&#10;Context: Page 0 (Main Settings)&#10;Item: frequency&#10;&#10;Place an underscore before the character that is to serve as the keyboard mnemonic.&#10;&#10;The mnemonic chosen for this item cannot be shared by any other item on this page&#10;(see context for translation) nor by any of the global context items!">Fre_quency of reports and actions</property>
                                 <property name="use_underline">True</property>
@@ -125,6 +126,7 @@
                                 <property name="width_request">490</property>
                                 <property name="height_request">55</property>
                                 <property name="visible">True</property>
+                                <property name="sensitive">False</property>
                                 <property name="xalign">0</property>
                                 <property name="yalign">0</property>
                                 <property name="label" translatable="yes" comments="Mnemonic: No&#10;Context: Page 0 (Main Settings)&#10;Item: delay description">Number of minutes to wait before waking up Prey. Control Panel users can change this setting later on the web.</property>
@@ -143,6 +145,7 @@
                                 <property name="width_request">48</property>
                                 <property name="height_request">48</property>
                                 <property name="visible">True</property>
+                                <property name="sensitive">False</property>
                                 <property name="pixbuf">../../pixmaps/conf/user.png</property>
                               </object>
                               <packing>
@@ -156,7 +159,8 @@
                                 <property name="width_request">200</property>
                                 <property name="height_request">30</property>
                                 <property name="visible">True</property>
-                                <property name="can_focus">True</property>
+                                <property name="can_focus">False</property>
+                                <property name="sensitive">False</property>
                                 <property name="receives_default">False</property>
                                 <property name="use_underline">True</property>
                                 <property name="draw_indicator">True</property>
@@ -171,6 +175,7 @@
                                 <property name="width_request">490</property>
                                 <property name="height_request">55</property>
                                 <property name="visible">True</property>
+                                <property name="sensitive">False</property>
                                 <property name="yalign">0</property>
                                 <property name="label" translatable="yes" comments="Mnemonic: No&#10;Context: Page 0 (Main Settings)&#10;Item: guest login description">Whether we should allow guest logins on the system. On password-protected computers, this greatly increases the chances of gathering information.</property>
                                 <property name="wrap">True</property>
